csrc/abstract_tensor.h

```diff
@@ -332,10 +849,139 @@ struct AbstractTensor {
   //   row2  id4    id5
   // in another word, the return value will be an AbstractTensor:
   // [dim0={id0, id1, id4}, dim1={id2, id3, id5}].
-  AbstractTensor& addRow(AbstractTensor tensor);
+  AbstractTensorWithInfo& addRow(AbstractTensorWithInfo tensor) {
+    NVF_CHECK(
+        domain.size() == tensor.domain.size(),
+        "Can not add a new row with different number of domains.");
+    NVF_CHECK(
+        std::all_of(
+            domain.begin(),
+            domain.end(),
+            [](const AbstractId& aid) { return aid.is<std::vector>(); }),
+        "Can not add a new row to an AbstractTensor with non-vector domains.")
+
+    for (auto i : c10::irange(domain.size())) {
+      domain[i].as<std::vector>().emplace_back(std::move(tensor.domain[i]));
```

Comment:
Should we do a
```C++
NVF_CHECK(return info[i] == tensor.info[i]);
result.info.emplace_back(std::move(tensor.info[i]));
```

Reply:
Yes to the first line, but we don't need to modify `info`, just check that it matches the new row.